amcl/CMakeLists.txt

```diff
@@ -1,141 +1,150 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(amcl)
 
-find_package(catkin REQUIRED
-        COMPONENTS
-            rosbag
-            roscpp
-            tf
-            dynamic_reconfigure
-            nav_msgs
-        )
+if(NOT WIN32)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wextra")
+endif()
 
-find_package(Boost REQUIRED)
+find_package(ament_cmake_auto REQUIRED)
+set(required_dependencies
+  "geometry_msgs"
+  "nav_msgs"
+  "tf2"
+  "tf2_ros"
+  "tf2_msgs"
+  "sensor_msgs"
+  "std_srvs"
+  "rclcpp"
+  "rcutils"
+  "tf2_geometry_msgs"
+)
+ament_auto_find_build_dependencies(REQUIRED ${required_dependencies})
 
 # dynamic reconfigure
-generate_dynamic_reconfigure_options(
-    cfg/AMCL.cfg
-)
+# generate_dynamic_reconfigure_options(
+#     cfg/AMCL.cfg
+# )
 
-catkin_package(
-    CATKIN_DEPENDS
-        rosbag
-        roscpp
-        dynamic_reconfigure
-        tf
-  INCLUDE_DIRS include
-  LIBRARIES amcl_sensors amcl_map amcl_pf
-)
 
 include_directories(include/amcl include/amcl/map include/amcl/sensors include/amcl/pf)
-include_directories(${catkin_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS})
-
-add_library(amcl_pf
+include_directories(
+                    ${rclcpp_INCLUDE_DIRS}
+                    ${tf2_INCLUDE_DIRS}
+                    ${tf2_msgs_INCLUDE_DIRS}
+                    ${tf2_geometry_msgs_INCLUDE_DIRS}
+                    ${tf2_ros_INCLUDE_DIRS}
+                    ${std_srvs_INCLUDE_DIRS}
+                    ${sensor_msgs_INCLUDE_DIRS}
+                    ${geometry_msgs_INCLUDE_DIRS}
+                    ${nav_msgs_INCLUDE_DIRS}
+                    )
```

Comment:
Because `ament_target_dependencies` does it for you AFAIK. Maybe that's what the "**TODO: explain how ament_target_dependencies() simplifies the above steps and is also better (also handling *_DEFINITIONS, doing target-specific include directories, etc.)** " is for.